subprojects/language-java/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -128,8 +129,11 @@ public JavaExec() {
         modularity = objectFactory.newInstance(DefaultModularitySpec.class);
         execResult = objectFactory.property(ExecResult.class);
         javaExecSpec = objectFactory.newInstance(DefaultJavaExecSpec.class);
+
         Provider<Iterable<String>> jvmArgumentsConvention = getProviderFactory()
             .provider(() -> getConventionMapping().getConventionValue(null, "jvmArgs", false));
+        jvmArgumentsConvention = jvmArgumentsConvention.orElse(new ArrayList<>());
```

Comment:
🔧 I think it would be clearer to have the whole _jvmArgs convention value_ computation in a single place. Also, it should be a simple `emptyList()` instead of a new `ArrayList` instance. Suggestions:
- Introduce a `jvmArgsConventionValue` method:
```java
    private Iterable<String> jvmArgsConventionValue() {
        Iterable<String> jvmArgs = getConventionMapping().getConventionValue(null, "jvmArgs", false);
        return jvmArgs != null ? jvmArgs : emptyList();
    }
```
- Create the convention from it:
```java
        Provider<Iterable<String>> jvmArgumentsConvention = getProviderFactory().provider(this::jvmArgsConventionValue);
```

Reply:
Okay, we are done